torch/_inductor/decomposition.py

```diff
@@ -591,6 +591,37 @@ def get_like_layout(
         return memory_format
 
 
+def _preserve_stride_pattern(
+    result: torch.Tensor,
+    reference: torch.Tensor,
+    memory_format: Optional[torch.memory_format] = None,
+) -> torch.Tensor:
+    """
+    Preserve the stride pattern of the reference tensor in the result tensor.
+
+    This is critical for maintaining the correct memory layout, especially for
+    non-contiguous tensors (e.g., after rot90).
+
+    Args:
+        result: The tensor to adjust
+        reference: The tensor whose stride pattern to match
+        memory_format: Optional memory format to apply
+
+    Returns:
+        The result tensor with matching stride pattern
+    """
+    # If memory_format is explicitly set, use it
+    if memory_format is not None and memory_format != torch.preserve_format:
+        return result.to(memory_format=memory_format)
+
+    # Otherwise, preserve the exact stride pattern of the reference tensor
+    if reference.stride() != result.stride():
+        # Use as_strided to match the exact stride pattern
+        result = torch.as_strided(result, reference.shape, reference.stride())
```

Comment:
I don't think this is correct in the case where there are gaps in the strides.  For example:
```py
>>> torch.rand_like(torch.empty_strided((10, 10, 10), (200, 20, 1))).stride()
(100, 10, 1)
>>> torch.rand_like(torch.empty_strided((10, 10, 10), (200, 1, 10))).stride()
(100, 1, 10)
>>> torch.rand_like(torch.empty_strided((10, 10, 10), (1, 20, 500))).stride()
(1, 10, 100)
```
It looks like eager preserves *order* but not exact values.